charts/gateway-helm/templates/certgen-rbac.yaml

```diff
@@ -1,3 +1,4 @@
+{{- if ne (toString .Values.certgen) "nil" }}
```

Comment:
shouldnt this be a `nil` `{{ if .Values.certgen }}` check rather than a string eq check ?

Reply:
https://helm.sh/docs/chart_template_guide/control_structures/#ifelse

Reply:
When we explicitly set `certgen: nil`, yaml does not consider this to be empty, so `{{ if .Values.certgen }}` will be executed incorrectly.

Reply:
Initially I also used the empty check, but it always failed. After many attempts, maybe helm treats yaml as a string when parsing it?

We can make a simple attempt:
```yaml
# values.yaml
certgen: nil

# template
{{- if eq .Values.certgen nil }}
msg: empty
{{- else }}
msg: no empty
{{- end }}
```
When we execute the `template` command, this will happen:
```shell
msg: no empty
```

Let's change another example
```yaml
# values.yaml
certgen: nil

# template
{{- if .Values.certgen }}
msg: no empty
{{- else }}
msg: empty
{{- end }}
```
After executing the `template` command:
```shell
msg: no empty
```

Reply:
changing `certgen:` to `None` or `{}` shall work

Reply:
for the context here, is `certgen` better to be a boolean ?

Reply:
In this context, I think it would be more appropriate to use this way:
```yaml
certgen:
# Add a knob to turn it on or off
enable: true/false
job:
annotations: {}
resources: {}
ttlSecondsAfterFinished: 30
rbac:
annotations: {}
labels: {}
```

Reply:
what is the most common usage in helm
* enable/disable
* enabled/disabled
* nil

Reply:
Combined with your comments. I think it is better to design it as follows:
```yaml
# This means disabling the Job
certgen: disable
```

@arkodg